### PR TITLE
Upgrades to GrandPotentialKernelAction

### LINKED ISSUE #13468: Minor improvements to GrandPotentialKernelAction
## Reason
In GrandPotentialKernelAction, the chemical potential mobilities currently cannot mix scalars and tensors. Also, an error has popped up with multiple chemical potentials not being linked in the CoupledSwitchingTimeDerivative kernels.

## Design
Add the capability to mix and match mobility data types and improve the linkage in CoupledSwitchingTimeDerivative kernels.

## Impact
This will improve the flexibility and performance of GrandPotentialKernelAction

### PR DESCRIPTION
**Pull request by itgreenquist**
Small updates to GrandPotentialKernelAction to improve coupling of variables and flexibility of anisotropic mobilities.

Closes #13468

### CODE REVIEW

File: modules/phase_field/src/action/GrandPotentialKernelAction.C [hunk 1]
```diff
@@ -65,7 +65,12 @@ validParams<GrandPotentialKernelAction>()
   parameters.addParam<bool>("implicit", true, "Whether kernels are implicit or not");
   parameters.addParam<bool>(
       "use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
-  parameters.addParam<bool>("anisotropic", false, "Set to true if the diffusivity is a tensor");
+  parameters.addRequiredParam<std::vector<int>>(
+      "anisotropy",
+      "Vector that determines if each difusivity associated with 'chemical_potentials' is "
+      "anisotropic. Set 0 for isotropic and 1 for anisotropic.");
+  // NOTE Moose does not support std::vector<bool>. This is the workaround proposed in
+  // https://groups.google.com/d/msg/moose-users/aroHLCG_8J8/0Zz6uwycCQAJ
```
Comment (dschwen) [inline modules/phase_field/src/action/GrandPotentialKernelAction.C]: A `MultiMooseEnum` would be a better choice here.

File: modules/phase_field/src/action/GrandPotentialKernelAction.C [hunk 2]
```diff
@@ -261,10 +272,15 @@ GrandPotentialKernelAction::act()
     params.set<bool>("use_displaced_mesh") = displaced_mesh;
     params.set<MaterialPropertyName>("D_name") = M[i];
     kernel_name = "MatDif_" + w_names[i];
-    if (aniso)
+    if (aniso[i])
+    {
       _problem->addKernel("MatAnisoDiffusion", kernel_name, params);
+    }
```
Comment (dschwen) [inline modules/phase_field/src/action/GrandPotentialKernelAction.C]: those braces are not necessary

File: modules/phase_field/src/action/GrandPotentialKernelAction.C [hunk 3]
```diff
@@ -248,7 +259,7 @@ GrandPotentialKernelAction::act()
     InputParameters params = _factory.getValidParams("SusceptibilityTimeDerivative");
     params.set<NonlinearVariableName>("variable") = w_names[i];
     params.set<MaterialPropertyName>("f_name") = chis[i];
-    params.set<std::vector<VariableName>>("args") = v1;
+    params.set<std::vector<VariableName>>("args") = v0;
```
Comment (dschwen) [inline modules/phase_field/src/action/GrandPotentialKernelAction.C]: why this change?

Comment (itgreenquist) [reply modules/phase_field/src/action/GrandPotentialKernelAction.C]: v0 contains chemical potentials as well as order parameters. This change allows it to couple chemical potential variables.

File: modules/phase_field/src/action/GrandPotentialKernelAction.C [hunk 4]
```diff
@@ -97,7 +99,7 @@ GrandPotentialKernelAction::act()
   const auto op_mob = getParam<MaterialPropertyName>("mobility_name_op");
   auto implicity = getParam<bool>("implicit");
   auto displaced_mesh = getParam<bool>("use_displaced_mesh");
-  auto aniso = getParam<bool>("anisotropic");
+  MultiMooseEnum aniso = getParam<MultiMooseEnum>("anisotropic");
```
Comment (dschwen) [inline modules/phase_field/src/action/GrandPotentialKernelAction.C]: keep the `auto`

File: modules/phase_field/src/action/GrandPotentialKernelAction.C [hunk 5]
```diff
@@ -65,7 +65,9 @@ validParams<GrandPotentialKernelAction>()
   parameters.addParam<bool>("implicit", true, "Whether kernels are implicit or not");
   parameters.addParam<bool>(
       "use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
-  parameters.addParam<bool>("anisotropic", false, "Set to true if the diffusivity is a tensor");
+  MultiMooseEnum anisotropy("true false", "false");
```
Comment (dschwen) [inline modules/phase_field/src/action/GrandPotentialKernelAction.C]: if you order this `false true` or explicitly set `true=1 false=0` you can use an implicit bool cast to check the value...

Comment (itgreenquist) [reply modules/phase_field/src/action/GrandPotentialKernelAction.C]: How would that look? On line 269 I have `if (aniso[i] == "true")`. If I reference `aniso[i]` it changes types to `std::string`, but `aniso.contains` function would search all of the entries, not just entry `i`.

Comment (itgreenquist) [reply modules/phase_field/src/action/GrandPotentialKernelAction.C]: Okay, I just found the `get` function.

**Comment by permcody** [comment]
@dschwen - Can you do a re-review? Thanks

File: modules/phase_field/src/action/GrandPotentialKernelAction.C [hunk 6]
```diff
@@ -118,6 +120,9 @@ GrandPotentialKernelAction::act()
                "switching_function_names");
   if (M.size() != n_w)
     mooseError("M and chemical_potentials should be vectors of the same length.");
+  if (aniso.size() != n_w)
+    mooseError(
```
Comment (dschwen) [inline modules/phase_field/src/action/GrandPotentialKernelAction.C]: I suggest `paramError("anisotropic", "Provide as many values as entries in 'chemical_potentials'.");`

Comment (dschwen) [reply modules/phase_field/src/action/GrandPotentialKernelAction.C]:
```suggestion
    paramError("anisotropic", "Provide as many values as entries in 'chemical_potentials'.");
```
